app/ante/priority.go

```diff
@@ -0,0 +1,70 @@
+package ante
+
+import (
+	"math"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	btcctypes "github.com/babylonlabs-io/babylon/x/btccheckpoint/types"
+	btclctypes "github.com/babylonlabs-io/babylon/x/btclightclient/types"
+	bstypes "github.com/babylonlabs-io/babylon/x/btcstaking/types"
+	bstctypes "github.com/babylonlabs-io/babylon/x/btcstkconsumer/types"
+	ftypes "github.com/babylonlabs-io/babylon/x/finality/types"
+)
+
+const (
+	// RegularTxMaxPriority is the max priority for transactions with regular messages.
+	// Thus, the reserved priority range for protocol liveness-related messages is (RegularTxMaxPriority, MaxInt64]
+	RegularTxMaxPriority = math.MaxInt64 - 1000
+	// LivenessTxPriority is the priority for protocol liveness-related messages.
+	// For the moment, the priority is the same for all of these messages
+	LivenessTxPriority = RegularTxMaxPriority + 100
+)
+
+// PriorityDecorator assigns higher priority to protocol liveness-related transactions
+type PriorityDecorator struct{}
+
+func NewPriorityDecorator() PriorityDecorator {
+	return PriorityDecorator{}
+}
+
+// Assigns higher priority to protocol liveness-related transactions
+func (pd PriorityDecorator) AnteHandle(ctx sdk.Context, tx sdk.Tx, simulate bool, next sdk.AnteHandler) (sdk.Context, error) {
+	// Cap priority for txs with regular messages
+	// Use higher priorities for liveness-related txs
+	priority := min(ctx.Priority(), RegularTxMaxPriority)
+
+	if isLivenessTx(tx) {
+		priority = LivenessTxPriority
+	}
+
+	newCtx := ctx.WithPriority(priority)
+
+	return next(newCtx, tx, simulate)
+}
+
+func isLivenessTx(tx sdk.Tx) bool {
+	for _, msg := range tx.GetMsgs() {
+		switch msg.(type) {
+		case *btclctypes.MsgInsertHeaders, // BTC light client
+			// BTC timestamping
+			*btcctypes.MsgInsertBTCSpvProof,
+			// BTC staking
+			*bstypes.MsgCreateFinalityProvider,
+			*bstypes.MsgCreateBTCDelegation,
```

Comment:
I do not think this should be prioritised  tbh

I think good rule of thumb which messages should have priority, is to prfioritize the ones which are currently being refunded (i.e there is call `IndexRefundableMsg` after message execution)

Maybe you can create small confluence page with all the messages that should be prioritized and then we can discuss it on the slack in async way ?

Reply:
doc [here](https://babylonlabs.atlassian.net/wiki/spaces/BABYLON/pages/140149031/Babylon+Liveness-Related+Transactions). Will post it in slack as well